api/public.proto

```diff
@@ -566,6 +588,64 @@ message ListTransfersReply {
   repeated Transfer transfers = 1;
 }
 
+message ListVehiclesRequest {
+  // ID of the system for which to list vehicles.
+  string system_id = 1;
+
+  enum SearchMode {
+    // Return a paginated list of stops sorted by stop ID.
+    ID = 0;
+    // Return all stops within max_distance of (latitude, longitude), sorted by the distance.
+    DISTANCE = 1;
+  }
+  // The type of search to perform when listing stops.
+  optional SearchMode search_mode = 2;
+
+  // If true, only return vehicles whose IDs are specified in the repeated `id` field.
+  // Only supported when the search mode is ID.
+  bool only_return_specified_ids = 3;
+
+  // IDs to return if `only_return_specified_ids` is set to true. It is an error to
+  // populate this field if `only_return_specified_ids` is false.
+  // Only supported when the search mode is ID.
+  repeated string id = 4;
+
+  // ID of the first stop to return. If not set, the stop with the smallest ID will be first.
+  // Only supported when the search mode is ID.
+  optional string first_id = 5;
+
+  // Maximum number of vehicles to return.
+  // This is supported in all search modes.
+  // For performance reasons, if it is larger than 100 it is rounded down to 100.
+  optional int32 limit = 6;
+
+  // The maximum distance in kilometers that a vehicle must be from
+  // latitude, longitude to be listed when using DISTANCE search mode.
+  optional double max_distance = 7;
+
+  // The latitude relative to the returned vehicles when using DISTANCE search mode.
+  optional double latitude = 8;
+
+  // The longitude relative to the returned vehicles when using DISTANCE search mode.
+  optional double longitude = 9;
+}
+
+message ListVehiclesReply {
+  // List of vehicles.
+  repeated Vehicle vehicles = 1;
+}
```

Comment:
I think we may need to add a `next_id` field to support pagination? This is me just blindly copying whatever is in `ListStopsReply`!

Reply:
Good catch, will fix and add some test coverage!